File: nomad-samples/python/basic-live-channel-1.1/live_input/get_live_input_status.py
from live_input.get_live_input import *

'''
 * Get Live Input Status
 *
 * @param {string} AUTH_TOKEN    | Authentication token
 * @param {string} INPUT_ID      | The INPUT ID for which to get the status
'''
async def get_live_input_status(AUTH_TOKEN, INPUT_ID):
    # Check for valid parameters
    if (not AUTH_TOKEN or not INPUT_ID):
        raise Exception("Get Live Input Status: Invalid API call")


    # Get the live INPUT
    INPUT = await get_live_input(AUTH_TOKEN, INPUT_ID)

    # Check if INPUT was found
    if (INPUT):
        # Return the status of the INPUT
        return INPUT["status"]["description"]


    # Input was not found
    return "Deleted"

File: nomad-samples/admin/js/basic-live-channel/live-input/get-live-input-status.js
import getLiveInput from "./get-live-input.js";

/**
 * Get Live Input Status
 *
 * @param {string} authToken    | Authorization token
 * @param {string} inputId      | The input ID for which to get the status
 */
export default async function getLiveInputStatus(authToken, inputId) {
    // Get the live input
    const input = await getLiveInput(authToken, inputId);

    // Check if input was found
    if (input) {
        // Return the status of the input
        return input.status.description;
    }

    // Input was not found
    return "Deleted";
}
